src/sqlfluff/dialects/dialect_materialize.py

```diff
@@ -0,0 +1,1016 @@
+            "INDEX",
+            Sequence(
+                "MATERIALIZED",
+                "VIEW"
+            ),
+            "ROLE",
+            "SECRET",
+            "SCHEMA",
+            "SINK",
+            "SOURCE",
+            "TABLE",
+            "TYPE",
+            "VIEW",
+            "USER"
+        ),
+        Ref("IfExistsGrammar", optional=True),
+        Ref("ObjectReferenceSegment"),
+        OneOf(
+            Sequence(
+                "CASCADE"
+            ),
+            Sequence(
+                "RESTRICT"
+            ),
+            optional=True
+        )
+    )
+
+class ShowStatementSegment(BaseSegment):
+    """A Materialize `SHOW` statement.
+    """
+    _object_types = OneOf(
+        "COLUMNS",
+        "CONNECTIONS",
+        "CLUSTERS",
+        Sequence("CLUSTER", "REPLICAS"),
+        "DATABASES",
+        "INDEXES",
+        Sequence("MATERIALIZED", "VIEWS"),
+        "SECRETS",
+        "SCHEMAS",
+        "SINKS",
+        "SOURCES",
+        "TABLES",
+        "TYPES",
+        "VIEWS",
+        "OBJECTS",
+    )
+
+    type = "show_statement"
+
+    match_grammar = Sequence(
+        "SHOW",
+        _object_types,
+        Ref("ObjectReferenceSegment", optional=True),
+        # FROM is optional for some object types
+        Sequence(
+            "FROM",
+            Ref("ObjectReferenceSegment"),
+            optional=True
+        ),
+        #  Like or where is optional for some object types
+        OneOf(
+            Sequence(
+                "LIKE",
+                Ref("QuotedLiteralSegment"),
+            ),
+            Sequence(
+                "WHERE",
+                Ref("ExpressionSegment"),
+            ),
+            optional=True
+        ),
+    )
+
+class ShowCreateStatementSegment(BaseSegment):
+    """A Materialize `SHOW CREATE` statement.
+    """
+    _object_types = OneOf(
+        Sequence("CONNECTION", optional=True),
+        Sequence("INDEX", optional=True),
+        Sequence("MATERIALIZED", "VIEW", optional=True),
+        Sequence("SINK", optional=True),
+        Sequence("SOURCE", optional=True),
+        Sequence("TABLE", optional=True),
+        Sequence("VIEW", optional=True),
+    )
+
+    type = "show_create_statement"
+
+    match_grammar = Sequence(
+        "SHOW",
+        "CREATE",
+        _object_types,
+        Ref("ObjectReferenceSegment")
+    )
+
+class ShowIndexesStatementSegment(BaseSegment):
+    """A Materialize `SHOW INDEXES` statement.
+    """
+
+    type = "show_indexes_statement"
+
+    match_grammar = Sequence(
+        "SHOW",
+        "INDEXES",
+        Sequence(
+            "ON",
+            Ref("ObjectReferenceSegment"),
+            optional=True
+        ),
+        Sequence(
+            "FROM",
+            Ref("ObjectReferenceSegment"),
+            optional=True
+        ),
+        Sequence(
+            "IN",
+            "CLUSTER",
+            Ref("ObjectReferenceSegment"),
+            optional=True
+        ),
```

Comment:
This thing seems to be repeated a few times. Maybe it should have its own grammar?

Reply:
Just updated this accordingly!